pilot/proxy/envoy/route.go

```diff
@@ -246,6 +282,158 @@ func buildHTTPRoute(config model.Config, service *model.Service, port *model.Por
 	return route
 }
 
+func buildHTTPRouteV2(config model.Config, service *model.Service, port *model.Port, instances []*model.ServiceInstance, domain string) []*HTTPRoute {
+	rule := config.Spec.(*routingv2.RouteRule)
+	routes := make([]*HTTPRoute, 0)
+
+	defaultDestination := service.Hostname
+	for _, http := range rule.Http {
+		matchRoutes := buildHTTPRouteMatches(instances, http.Match)
+		for _, route := range matchRoutes {
+			// TODO: logic in this block is not dependent on the route match, so we may be able to avoid
+			// rerunning it N times
+
+			if len(http.Route) == 0 { // build default cluster
+				cluster := buildOutboundCluster(defaultDestination, port, nil)
+				route.Cluster = cluster.Name
+				route.clusters = append(route.clusters, cluster)
+			} else {
+				route.WeightedClusters = &WeightedCluster{Clusters: make([]*WeightedClusterEntry, 0, len(http.Route))}
+				for _, dst := range http.Route {
+					fqdn := model.ResolveFQDN(dst.Destination.Name, domain)
+					cluster := buildOutboundCluster(fqdn, port, dst.Destination.Labels) // TODO: support Destination.Port
```

Comment:
This is a big todo.. is it the cae that we are not supplying port information in the registration key?

Reply:
I'll create an issue to track this